primitives/src/opcodes.rs

```diff
@@ -549,7 +583,13 @@ mod tests {
             assert_eq!($op, Opcode::from($op.to_u8()));
 
             let s1 = format!("{}", $op);
+            let parsed = s1.parse::<Opcode>().expect("failed to parse opcode (Display)");
+            assert_eq!(parsed, $op);
+
             let s2 = format!("{:?}", $op);
+            let parsed = s1.parse::<Opcode>().expect("failed to parse opcode (Debug)");
+            assert_eq!(parsed, $op);
+
```

Comment:
I am reimplementing this in #3729.
But just for the sake of correctness if someone or something uses this close PR in the future.
The second `parsed` should be `s2` and not `s1`

```suggestion
            let parsed = s1.parse::<Opcode>().expect("failed to parse opcode (Display)");
            assert_eq!(parsed, $op);
            let s2 = format!("{:?}", $op);
            let parsed = s2.parse::<Opcode>().expect("failed to parse opcode (Debug)");
            assert_eq!(parsed, $op);

```